crates/eww/src/config/system_stats.rs

```diff
@@ -72,16 +78,15 @@ pub fn get_ram() -> String {
 
 pub fn get_temperatures() -> String {
     let mut components = COMPONENTS.lock().unwrap();
-    components.refresh_list();
-    components.refresh();
+    components.refresh(true);
     components
         .iter()
         .map(|c| {
             (
                 c.label().to_uppercase().replace(' ', "_"),
                 // It is common for temperatures to report a non-numeric value.
                 // Tolerate it by serializing it as the string "null"
-                c.temperature().to_string().replace("NaN", "\"null\""),
+                c.temperature().unwrap_or(NAN).to_string().replace("NaN", "\"null\""),
```

Comment:
this should leverage the power of the type system.
consider this, from #1217:
```suggestion
                c.temperature().map_or_else(|| String::from("\"null\""), |temp| temp.to_string()),
```